Document that `parse` runs when real-time grading is disabled (#12242)
PrairieLearn/PrairieLearn: docs/assessment/index.md

An assessment without real-time grading will not show any score information during the exam. However, if a [time limit](../accessControl/index.md#time-limits) is used then when it runs out the assessment will auto-grade and show students exactly which questions they got correct/incorrect. The same revealing behavior will happen if an instructor manually closes and grades the student assessment. To prevent this, set the [`showClosedAssessment` access rule restriction](../accessControl/index.md#showinghiding-closed-assessments).

??? note

>     Submissions will still be parsed and in particular when students click the "Save" button, they will be informed of invalid submissions (as usual, this does not count as an attempt).

Reviewer: The `parse` function in server.py is only part of the parse process. Elements also parse the values. I'd suggest something in the direction of: 

```suggestion
    Submissions will still be parsed and in particular when students click the "Save" button, they will be informed of invalid submissions. This includes any pre-processing of content associated to individual elements (e.g., checking for blank answers, or checking if numeric inputs have numeric values), as well as [question-specific custom parse functionality](../question/server.md#step-4-parse). As usual, invalid submissions do not count towards limited attempts, and are not graded.
```

>     Students will not be prevented from completing the test with invalid submissions should they choose to do so, or if they run out of time before crafting a valid submission.

Reviewer: ```suggestion
    Students will not be prevented from completing the test with invalid submissions should they choose to do so, or if they run out of time before crafting a valid submission. However, invalid submissions will not trigger any auto-grading process, and will by default cause the student to get a zero in the whole question (even if there are elements in the submission with valid answers). Manual grading is still available for these submissions.
```

An assessment without real-time grading will not show any score information during the exam. However, if a [time limit](../accessControl/index.md#time-limits) is used then when it runs out the assessment will auto-grade and show students exactly which questions they got correct/incorrect. The same revealing behavior will happen if an instructor manually closes and grades the student assessment. To prevent this, set the [`showClosedAssessment` access rule restriction](../accessControl/index.md#showinghiding-closed-assessments).

> Note that submissions will still be parsed and in particular when students click the "Save" button, they will be informed of invalid submissions (and as usual, this does not count as an attempt).

Reviewer 2: Break out to an admonition.

```suggestion
??? note

    Submissions will still be parsed and in particular when students click the "Save" button, they will be informed of invalid submissions (and as usual, this does not count as an attempt).
```
